testing/test-utils/src/lib/utils/os-agnostic-paths.unit.test.ts

```diff
@@ -0,0 +1,121 @@
+      expect(
+        osAgnosticPath(`${linuxmMacOSCWD}/.code-pushup/.code-pushup.config.ts`),
+      ).toBe('<CWD>/.code-pushup/.code-pushup.config.ts');
+    });
+
+    it('should return paths outside of CWD on Linux/macOS', () => {
+      expect(
+        osAgnosticPath(
+          `${linuxmMacOSCWD}/../.code-pushup/.code-pushup.config.ts`,
+        ),
+      ).toBe('../.code-pushup/.code-pushup.config.ts');
+    });
+
+    it('should handle absolute paths correctly on Linux/macOS', () => {
+      expect(osAgnosticPath('/.code-pushup/.code-pushup.config.ts')).toBe(
+        '/.code-pushup/.code-pushup.config.ts',
+      );
+    });
+
+    it('should handle paths with CWD shorthand "." correctly on Linux/macOS', () => {
+      expect(osAgnosticPath('./.code-pushup/.code-pushup.config.ts')).toBe(
+        './.code-pushup/.code-pushup.config.ts',
+      );
+    });
+
+    it('should handle relative paths correctly on Linux/macOS', () => {
+      expect(osAgnosticPath('../../.code-pushup/.code-pushup.config.ts')).toBe(
+        '../../.code-pushup/.code-pushup.config.ts',
+      );
+    });
+
+    it('should handle path segments correctly on Linux/macOS', () => {
+      expect(osAgnosticPath('.code-pushup/.code-pushup.config.ts')).toBe(
+        '.code-pushup/.code-pushup.config.ts',
+      );
+    });
+
+    it('should NOT modify already OS-agnostic paths on Linux/macOS', () => {
+      expect(osAgnosticPath('<CWD>/.code-pushup/.code-pushup.config.ts')).toBe(
+        '<CWD>/.code-pushup/.code-pushup.config.ts',
+      );
+    });
+  });
+
+  describe('Windows', () => {
+    const windowsCWD = 'D:\\users\\jerry';
+
+    beforeEach(() => {
+      cwdSpy.mockReturnValue(windowsCWD);
+    });
+    afterEach(() => {
+      cwdSpy.mockReset();
+    });
+
+    it('should return paths outside of CWD on Windows', () => {
+      expect(
+        osAgnosticPath(
+          `${windowsCWD}\\..\\.code-pushup\\.code-pushup.config.ts`,
+        ),
+      ).toBe('../.code-pushup/.code-pushup.config.ts');
+    });
+
+    it('should convert a path within the CWD to an OS-agnostic path on Windows', () => {
+      expect(
+        osAgnosticPath(`${windowsCWD}\\.code-pushup\\.code-pushup.config.ts`),
+      ).toBe('<CWD>/.code-pushup/.code-pushup.config.ts');
+    });
+
+    it('should handle absolute paths correctly on Windows', () => {
+      expect(osAgnosticPath('\\.code-pushup\\.code-pushup.config.ts')).toBe(
+        '/.code-pushup/.code-pushup.config.ts',
+      );
+    });
+
+    it('should handle paths with CWD shorthand "." correctly on Windows', () => {
+      expect(osAgnosticPath('./.code-pushup/.code-pushup.config.ts')).toBe(
+        './.code-pushup/.code-pushup.config.ts',
+      );
+    });
+
+    it('should handle relative paths correctly on Windows', () => {
+      expect(osAgnosticPath('../../.code-pushup/.code-pushup.config.ts')).toBe(
+        '../../.code-pushup/.code-pushup.config.ts',
+      );
+    });
```

Comment:
Shouldn't you be using backslashes here in the input?

Reply:
fixed! nice catche!